### ISSUE TITLE
Consider making sbt less verbose

### ISSUE THREAD

**Issue by jvican** [sbt/sbt#2994]
Current sbt is extremely verbose. Messages could be improved to be:
  
* More concrete. Straight to the point.
* Shorter. This also includes removing unnecessary messages.
* Grouped by topic (perhaps with progress bars).
  
I'm going to illustrate my points with examples.
  
Very short extract of `publishLocal` in `sbt/sbt`:
```

[info] will publish locally with javac version 1.8.0_92
[info] will publish locally with javac version 1.8.0_92
[info] :: delivering :: org.scala-sbt#test-agent;0.13.14-SNAPSHOT :: 0.13.14-SNAPSHOT :: integration :: Mon Mar 06 15:03:11 CET 2017
[info] 	delivering ivy file to /data/rw/code/scala/sbt/testing/agent/target/ivy-0.13.14-SNAPSHOT.xml
[info] :: delivering :: org.scala-sbt#collections;0.13.14-SNAPSHOT :: 0.13.14-SNAPSHOT :: integration :: Mon Mar 06 15:03:11 CET 2017
[info] 	delivering ivy file to /data/rw/code/scala/sbt/util/collection/target/ivy-0.13.14-SNAPSHOT.xml
[warn] bootstrap class path not set in conjunction with -source 1.6
[info] will publish locally with javac version 1.8.0_92
[info] :: delivering :: org.scala-sbt#cross;0.13.14-SNAPSHOT :: 0.13.14-SNAPSHOT :: integration :: Mon Mar 06 15:03:12 CET 2017
[info] 	delivering ivy file to /data/rw/code/scala/sbt/util/cross/target/ivy-0.13.14-SNAPSHOT.xml
[info] will publish locally with javac version 1.8.0_92
[info] :: delivering :: org.scala-sbt#sbt-launch;0.13.14-SNAPSHOT :: 0.13.14-SNAPSHOT :: integration :: Mon Mar 06 15:03:12 CET 2017
[info] 	delivering ivy file to /data/rw/code/scala/sbt/launch/target/ivy-0.13.14-SNAPSHOT.xml
[info] Packaging /data/rw/code/scala/sbt/util/collection/target/collections-0.13.14-SNAPSHOT.jar ...
[info] Done packaging.
[info] will publish locally with javac version 1.8.0_92
[info] :: delivering :: org.scala-sbt#control;0.13.14-SNAPSHOT :: 0.13.14-SNAPSHOT :: integration :: Mon Mar 06 15:03:12 CET 2017
[info] 	delivering ivy file to /data/rw/code/scala/sbt/util/control/target/ivy-0.13.14-SNAPSHOT.xml
[info] Packaging /data/rw/code/scala/sbt/util/cross/target/cross-0.13.14-SNAPSHOT.jar ...
[info] Done packaging.
[info] will publish locally with javac version 1.8.0_92
[info] :: delivering :: org.scala-sbt#process;0.13.14-SNAPSHOT :: 0.13.14-SNAPSHOT :: integration :: Mon Mar 06 15:03:12 CET 2017
[info] 	delivering ivy file to /data/rw/code/scala/sbt/util/process/target/ivy-0.13.14-SNAPSHOT.xml
[info] 	published collections to /home/jvican/.ivy2/local/org.scala-sbt/collections/0.13.14-SNAPSHOT/jars/collections.jar
[info] 	published collections to /home/jvican/.ivy2/local/org.scala-sbt/collections/0.13.14-SNAPSHOT/srcs/collections-sources.jar
[info] 	published ivy to /home/jvican/.ivy2/local/org.scala-sbt/collections/0.13.14-SNAPSHOT/ivys/ivy.xml
[info] 	published cross to /home/jvican/.ivy2/local/org.scala-sbt/cross/0.13.14-SNAPSHOT/jars/cross.jar
[info] 	published cross to /home/jvican/.ivy2/local/org.scala-sbt/cross/0.13.14-SNAPSHOT/srcs/cross-sources.jar
[info] 	published ivy to /home/jvican/.ivy2/local/org.scala-sbt/cross/0.13.14-SNAPSHOT/ivys/ivy.xml
[info] Packaging /data/rw/code/scala/sbt/util/control/target/control-0.13.14-SNAPSHOT.jar ...
[info] Done packaging.
[info] 	published sbt-launch to /home/jvican/.ivy2/local/org.scala-sbt/sbt-launch/0.13.14-SNAPSHOT/jars/sbt-launch.jar
[info] 	published ivy to /home/jvican/.ivy2/local/org.scala-sbt/sbt-launch/0.13.14-SNAPSHOT/ivys/ivy.xml
[info] Packaging /data/rw/code/scala/sbt/util/process/target/process-0.13.14-SNAPSHOT.jar ...
[info] will publish locally with javac version 1.8.0_92
[info] Done packaging.
[info] :: delivering :: org.scala-sbt#apply-macro;0.13.14-SNAPSHOT :: 0.13.14-SNAPSHOT :: integration :: Mon Mar 06 15:03:13 CET 2017
[info] 	delivering ivy file to /data/rw/code/scala/sbt/util/appmacro/target/ivy-0.13.14-SNAPSHOT.xml
[info] 	published control to /home/jvican/.ivy2/local/org.scala-sbt/control/0.13.14-SNAPSHOT/jars/control.jar
[info] 	published control to /home/jvican/.ivy2/local/org.scala-sbt/control/0.13.14-SNAPSHOT/srcs/control-sources.jar
[info] 	published ivy to /home/jvican/.ivy2/local/org.scala-sbt/control/0.13.14-SNAPSHOT/ivys/ivy.xml
[info] 	published process to /home/jvican/.ivy2/local/org.scala-sbt/process/0.13.14-SNAPSHOT/jars/process.jar
[info] 	published process to /home/jvican/.ivy2/local/org.scala-sbt/process/0.13.14-SNAPSHOT/srcs/process-sources.jar
[info] 	published ivy to /home/jvican/.ivy2/local/org.scala-sbt/process/0.13.14-SNAPSHOT/ivys/ivy.xml
[info] Packaging /data/rw/code/scala/sbt/util/appmacro/target/apply-macro-0.13.14-SNAPSHOT.jar ...
[info] Done packaging.
[info] will publish locally with javac version 1.8.0_92
[info] :: delivering :: org.scala-sbt#io;0.13.14-SNAPSHOT :: 0.13.14-SNAPSHOT :: integration :: Mon Mar 06 15:03:13 CET 2017
[info] 	delivering ivy file to /data/rw/code/scala/sbt/util/io/target/ivy-0.13.14-SNAPSHOT.xml
[info] will publish locally with javac version 1.8.0_92
[info] 	published apply-macro to /home/jvican/.ivy2/local/org.scala-sbt/apply-macro/0.13.14-SNAPSHOT/jars/apply-macro.jar
[info] 	published apply-macro to /home/jvican/.ivy2/local/org.scala-sbt/apply-macro/0.13.14-SNAPSHOT/srcs/apply-macro-sources.jar
[info] 	published ivy to /home/jvican/.ivy2/local/org.scala-sbt/apply-macro/0.13.14-SNAPSHOT/ivys/ivy.xml
[info] :: delivering :: org.scala-sbt#tasks;0.13.14-SNAPSHOT :: 0.13.14-SNAPSHOT :: integration :: Mon Mar 06 15:03:13 CET 2017
[info] 	delivering ivy file to /data/rw/code/scala/sbt/tasks/target/ivy-0.13.14-SNAPSHOT.xml
[info] Packaging /data/rw/code/scala/sbt/tasks/target/tasks-0.13.14-SNAPSHOT.jar ...
[info] Packaging /data/rw/code/scala/sbt/util/io/target/io-0.13.14-SNAPSHOT.jar ...
[info] Done packaging.
[info] Packaging /data/rw/code/scala/sbt/testing/agent/target/test-agent-0.13.14-SNAPSHOT.jar ...
[info] Done packaging.
[info] Done packaging.
[info] Compiling 4 Scala sources to /data/rw/code/scala/sbt/cache/target/classes...
[info] 	published tasks to /home/jvican/.ivy2/local/org.scala-sbt/tasks/0.13.14-SNAPSHOT/jars/tasks.jar
[info] 	published tasks to /home/jvican/.ivy2/local/org.scala-sbt/tasks/0.13.14-SNAPSHOT/srcs/tasks-sources.jar
[info] 	published ivy to /home/jvican/.ivy2/local/org.scala-sbt/tasks/0.13.14-SNAPSHOT/ivys/ivy.xml
[info] Compiling 2 Scala sources to /data/rw/code/scala/sbt/util/complete/target/classes...
[info] 	published test-agent to /home/jvican/.ivy2/local/org.scala-sbt/test-agent/0.13.14-SNAPSHOT/jars/test-agent.jar
[info] 	published test-agent to /home/jvican/.ivy2/local/org.scala-sbt/test-agent/0.13.14-SNAPSHOT/srcs/test-agent-sources.jar
[info] 	published ivy to /home/jvican/.ivy2/local/org.scala-sbt/test-agent/0.13.14-SNAPSHOT/ivys/ivy.xml
[info] will publish locally with javac version 1.8.0_92
[info] will publish locally with javac version 1.8.0_92
[info] will publish locally with javac version 1.8.0_92
[info] :: delivering :: org.scala-sbt#datatype-generator;0.13.14-SNAPSHOT :: 0.13.14-SNAPSHOT :: integration :: Mon Mar 06 15:03:13 CET 2017
[info] 	delivering ivy file to /data/rw/code/scala/sbt/util/datatype/target/ivy-0.13.14-SNAPSHOT.xml
[info] :: delivering :: org.scala-sbt#completion;0.13.14-SNAPSHOT :: 0.13.14-SNAPSHOT :: integration :: Mon Mar 06 15:03:13 CET 2017
[info] 	delivering ivy file to /data/rw/code/scala/sbt/util/complete/target/ivy-0.13.14-SNAPSHOT.xml
[info] :: delivering :: org.scala-sbt#cache;0.13.14-SNAPSHOT :: 0.13.14-SNAPSHOT :: integration :: Mon Mar 06 15:03:13 CET 2017
[info] 	delivering ivy file to /data/rw/code/scala/sbt/cache/target/ivy-0.13.14-SNAPSHOT.xml
[info] will publish locally with javac version 1.8.0_92
[info] 	published io to /home/jvican/.ivy2/local/org.scala-sbt/io/0.13.14-SNAPSHOT/jars/io.jar
[info] 	published io to /home/jvican/.ivy2/local/org.scala-sbt/io/0.13.14-SNAPSHOT/srcs/io-sources.jar
[info] 	published ivy to /home/jvican/.ivy2/local/org.scala-sbt/io/0.13.14-SNAPSHOT/ivys/ivy.xml
[info] :: delivering :: org.scala-sbt#scripted-framework;0.13.14-SNAPSHOT :: 0.13.14-SNAPSHOT :: integration :: Mon Mar 06 15:03:13 CET 2017
[info] 	delivering ivy file to /data/rw/code/scala/sbt/scripted/base/target/ivy-0.13.14-SNAPSHOT.xml
[info] Packaging /data/rw/code/scala/sbt/scripted/base/target/scripted-framework-0.13.14-SNAPSHOT.jar ...
[info] Done packaging.
[info] Packaging /data/rw/code/scala/sbt/util/datatype/target/datatype-generator-0.13.14-SNAPSHOT.jar ...
[info] Done packaging.
[info] 	published scripted-framework to /home/jvican/.ivy2/local/org.scala-sbt/scripted-framework/0.13.14-SNAPSHOT/jars/scripted-framework.jar
[info] 	published scripted-framework to /home/jvican/.ivy2/local/org.scala-sbt/scripted-framework/0.13.14-SNAPSHOT/srcs/scripted-framework-sources.jar
[info] 	published ivy to /home/jvican/.ivy2/local/org.scala-sbt/scripted-framework/0.13.14-SNAPSHOT/ivys/ivy.xml
[info] 	published datatype-generator to /home/jvican/.ivy2/local/org.scala-sbt/datatype-generator/0.13.14-SNAPSHOT/jars/datatype-generator.jar
[info] 	published datatype-generator to /home/jvican/.ivy2/local/org.scala-sbt/datatype-generator/0.13.14-SNAPSHOT/srcs/datatype-generator-sources.jar
[info] 	published ivy to /home/jvican/.ivy2/local/org.scala-sbt/datatype-generator/0.13.14-SNAPSHOT/ivys/ivy.xml
[info] will publish locally with javac version 1.8.0_92
[info] :: delivering :: org.scala-sbt#interface;0.13.14-SNAPSHOT :: 0.13.14-SNAPSHOT :: integration :: Mon Mar 06 15:03:14 CET 2017
[info] 	delivering ivy file to /data/rw/code/scala/sbt/interface/target/ivy-0.13.14-SNAPSHOT.xml
[info] Writing version information to /data/rw/code/scala/sbt/interface/target/resource_managed/xsbt.version.properties :
[info] version=0.13.14-SNAPSHOT
[info] timestamp=20170306T140314
[info] Packaging /data/rw/code/scala/sbt/interface/target/interface-0.13.14-SNAPSHOT.jar ...
[info] Done packaging.
[info] 	published interface to /home/jvican/.ivy2/local/org.scala-sbt/interface/0.13.14-SNAPSHOT/jars/interface.jar
[info] 	published interface to /home/jvican/.ivy2/local/org.scala-sbt/interface/0.13.14-SNAPSHOT/srcs/interface-sources.jar
[info] 	published ivy to /home/jvican/.ivy2/local/org.scala-sbt/interface/0.13.14-SNAPSHOT/ivys/ivy.xml
```
  
To me, those error messages look unhygienic and clutter my screen. I cannot focus on what sbt does.
  
What I'd like is:
  
* Remove deduplicated error messages (see how many times "using javac version" appears).
* Make messages more concise. Most of the error messages in that report are unnecessary. Why do I care where a xml file is published? And whether sbt writes or not the version to a file? Most of the times I don't. If I want that information, I'd like to pass in a verbose flag that does it.
  
We see these messages bazillion of times a day. It's time that we optimize them for the most common use cases and unclutter our workspaces.
  
Proposed error message for `publish*`:
```
[info] Using javac version 1.8.0_92.
[info] Packaging sbt-util.control-0.13.14-SNAPSHOT.jar ...
[success] Packaged artifacts can be found in  /data/rw/code/scala/sbt/util/control/target/.
[success] Published sbt-util to /home/jvican/.ivy2/local/org.scala-sbt/sbt-util/0.13.14-SNAPSHOT/.
```

**Comment by dwijnand** [comment]
Ah, the "it's all spam" vs "I've no idea what it's doing" endless debate..

Personally I've become numb to the verbosity, but I would welcome changes that lowered their visibility (e.g they would only be visible in `last`).

I don't think we should outright remove the messages, or at least not without a strong argument as to why.

:+1:

**Comment by jvican** [comment]
> but I would welcome changes that lowered their visibility (e.g they would only be visible in last).
  
Yes, I agree they should not be thrown away. I'd like to pass in a verbose flag, as I said in the previous message, or yust make them visible to `last`. I think this change will be very welcome from the community. After discussing with several people, I'm not the only one that thinks sbt should be less verbose.

**Comment by jvican** [comment]
Another thing I'd like to propose to current maintainers: drop logs like "Done X". I think that it should be implicit the fact that a task has finished when another task starts. Example:
  
```
Updating blabla.
Done updating.
Compiling $blabla.
Done compiling.
```
  
We should just remove the "Done" sentences since they don't bring anything to the table and make sbt output larger than it should, IMO.

**Comment by olafurpg** [comment]
I think the approach in #3122 is good, demote the severity of some verbose statements from info to debug. That's less radical than dropping logs.

**Comment by jvican** [comment]
> We should just remove the "Done" sentences since they don't bring anything to the table and make sbt output larger than it should, IMO.
  
Totally agree. @olafurpg Yes, I like the direction of that PR.

**Comment by SethTisue** [comment]
> I think that it should be implicit the fact that a task has finished when another task starts

tasks run in parallel

**Comment by jvican** [comment]
@SethTisue not necessarily, tasks can be run sequentially. It depends on your build graph.

**Comment by eed3si9n** [comment]
Closing this in favor of https://github.com/sbt/sbt/issues/4362
